core/state/statedb.go

```diff
@@ -474,24 +474,20 @@ func (s *StateDB) SetStorage(addr common.Address, storage map[common.Hash]common
 	}
 }
 
-// Suicide marks the given account as suicided.
+// SelfDestruct marks the given account as selfdestructed.
 // This clears the account balance.
 //
 // The account's state object is still available until the state is committed,
-// getStateObject will return a non-nil account after Suicide.
-func (s *StateDB) Suicide(addr common.Address) bool {
+// getStateObject will return a non-nil account after SelfDestruct.
+func (s *StateDB) SelfDestruct(addr common.Address) {
 	stateObject := s.getStateObject(addr)
-	if stateObject == nil {
-		return false
-	}
-	s.journal.append(suicideChange{
+	s.journal.append(selfDestructChange{
```

Comment:
I dunno -- all other places where we call `s.getStateObject` for various purposes, we do the nil-check. Seems pretty cheap to keep it, even if it might be un-needed in this particular path.